### DISCUSSION TITLE
How to delete the storage of a textEditor

### DISCUSSION THREAD

**Discussion by AlexGalays** [liveblocks/liveblocks#2094] [Help] [answered]
Hello 👋 
Currently evaluating Liveblocks!

Our app has a very dynamic number of text editors (you could have hundreds; is that even advised? I'm using `useLiveblocksExtension` from `react-tiptap`). Once I remove the react component hosting a text editor instance, it would be good to delete the data it had in liveblocks's storage so that the next time I recycle its ID (via the `field` option), I can start from scratch again (using the provided `initialContent`)

Is there any way to do that today?

**Comment by nvie** [comment]
Thanks for writing this up, @AlexGalays. Hmm, we do have a [Delete Storage document endpoint](https://liveblocks.io/docs/api-reference/rest-api-endpoints#delete-rooms-roomId-storage), but I just realize we don't offer a similar endpoint for Yjs yet. @jrowny Should we add that endpoint? Any advice for a workaround in the mean time?

**Reply by jrowny** [reply, marked as answer]
Best way is just to call editor.commands.clearContent() before un rendering the component.

Yjs doesn't really have a concept of deleting top level fields unless they're in a map. A possible enhancement for our editor is to add a fragment option instead of just field and then you can supply a fragment that's part of a map.

**Reply by jrowny** [reply]
And we don't offer a delete storage endpoint for yjs because clients would need to know to throw away their local ydoc, otherwise they'd potentially just sync their content back into the empty ydoc on the server.
